R/md_document.R

```diff
@@ -60,6 +60,19 @@ md_document <- function(variant = "markdown_strict",
   # pandoc args
   args <- c(args, pandoc_args)
 
+  # add pre_processor to fix link IDs when using numbering sections
+  pre_processor <- if (number_sections) {function(metadata, input_file, ...) {
+    markdown_format <- md_document(
+      variant = "markdown",
+      preserve_yaml = TRUE,
+      pandoc_args = c("--metadata", "preprocess_number_sections=true")
+    )
+    markdown_format$pandoc$lua_filters <- pkg_file_lua("number-sections.lua")
+    render(input_file, markdown_format, quiet = TRUE)
+    warning(paste(xfun::read_utf8(input_file), collapse="\n"))
```

Comment:
What's this `warning()` for?

Reply:
Oops, sorry that comes from my debugging

Reply:
No worries. You are not alone: https://twitter.com/xieyihui/status/1382100305588932608

Next time you could try something shorter like this :)

```r
warning(xfun::file_string(input_file))
```

Reply:
Wpw, thanks for your tip!

Reply:
Or if warnings are suppressed somehow, you may try `cat(x, file = stderr())`. No one can suppress our debugging messages!!